### PR TITLE
add operator dynamic class attribute

### PR DESCRIPTION
**Pull request by geospatial-jeff**

### CODE REVIEW

File: stac_pydantic/api/extensions/query.py [hunk 1]
```diff
@@ -1,7 +1,13 @@
+import operator
 from enum import auto
+from types import DynamicClassAttribute
+from typing import Any, Callable
 
 from stac_pydantic.utils import AutoValueEnum
 
+# TODO: These are defined in the spec but aren't currently implemented by the operator syntax
+UNSUPPORTED_OPERATORS = {"startsWith", "endsWith", "contains", "in"}
```
Comment (geospatial-jeff) [inline stac_pydantic/api/extensions/query.py:9]: these fields straight up don't work with python's `operator` 🤷‍♂️

Comment (geospatial-jeff) [reply stac_pydantic/api/extensions/query.py:9]: tbh I kind of hope CQL makes them irrelevent

Comment (vincentsarago) [reply stac_pydantic/api/extensions/query.py:9]: maybe we don't use the operator module 
```python
Operators = dict(
    eq = lambda x, y: x == y,
    ne = lambda x, y: x != y,
    lt = lambda x, y: x < y,
    le = lambda x, y: x <= y,
    gt = lambda x, y: x > y,
    ge = lambda x, y: x > y,
    startsWith = lambda x, y: x.startsWith(y),
    endsWith = lambda x, y: x.endsWith(y),
    constains = lambda x, y: y in x,
)
```

File: stac_pydantic/api/extensions/query.py [hunk 2]
```diff
@@ -8,6 +7,18 @@
 # TODO: These are defined in the spec but aren't currently implemented by the operator syntax
 UNSUPPORTED_OPERATORS = {"startsWith", "endsWith", "contains", "in"}
 
+_OPERATIONS = {
+    "eq": lambda x, y: x == y,
+    "ne": lambda x, y: x != y,
+    "lt": lambda x, y: x < y,
+    "le": lambda x, y: x <= y,
+    "gt": lambda x, y: x > y,
+    "ge": lambda x, y: x > y,
```
Comment (vincentsarago) [inline stac_pydantic/api/extensions/query.py:16]: you shouldn't copy code I write @geospatial-jeff 

```
"ge": lambda x, y: x > y,
```

**Review by lossyrob — APPROVED** [review]
This change seems good, there's a TODO in there but as per the comments I believe the upcoming CQL changes will deprioritize the need for full operator coverage.

**Pull request merged**